src/test/java/com/github/codeidoscope/ServerTest.java

```diff
@@ -0,0 +1,69 @@
+package com.github.codeidoscope;
+
+
+import org.junit.jupiter.api.Test;
+
+import java.io.BufferedReader;
+import java.io.PrintWriter;
+import java.io.StringReader;
+import java.io.StringWriter;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class ServerTest {
+
+    private StringReader stringReader;
+    private BufferedReader bufferedReader;
+    private StringWriter stringWriter;
+    PrintWriter printWriter;
+    private MockServerConnection mockServerConnection;
+    Server server;
+
+    void setUp(String input) {
+        stringReader = new StringReader(input);
+        bufferedReader = new BufferedReader(stringReader);
+        stringWriter = new StringWriter();
+        printWriter = new PrintWriter(stringWriter);
+        mockServerConnection = new MockServerConnection(bufferedReader, printWriter);
+        server = new Server(mockServerConnection);
+    }
+
+    @Test
+     void testAServerRespondsWithPoloWentUserSendsMarco() {
+        setUp("");
```

Comment:
You can let [JUnit do this for you](https://junit.org/junit5/docs/5.0.2/api/org/junit/jupiter/api/BeforeEach.html). Annotate the `setUp` method with `@BeforeEach` and the method will be invoked for you before each test.

Usually when writing a setup function for a test class, it should almost always be paired with an `@AfterEach` to clean the slate for the next test.

Reply:
I originally had a `@BeforeEach`, but in order to keep it a little cleaner while allowing for flexibility, I updated the `setUp` method to be a `@BeforeEach` so that it can take a string which can be changed in the test (the last two tests take different strings).

With a regular `@BeforeEach`, I am not able to manipulate it so easily, as I create a `StringReader`, which takes that string as an input, and which the rest of my setup depends on, so I could not create the `StringReader` separately as it cannot be created after the `@BeforeEach`.

If there's an elegant way to handle this in a `@BeforeEach`, please let me know, but it looked as though using the `@BeforeEach` would involve more work than just having a `setUp` method.

Reply:
👍